src/Application/UseCases/SetApproverCategories.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\PageApprovals\Application\UseCases;
+
+use ProfessionalWiki\PageApprovals\Adapters\DatabaseApproverRepository;
+
+class SetApproverCategories {
+
+	public function __construct(
+		private readonly DatabaseApproverRepository $databaseApproverRepository
+	) {
+	}
+
+	/**
+	 * @param string[] $categories
+	 */
+	public function setApproverCategories( int $userId, array $categories ): void {
+		$this->databaseApproverRepository->setApproverCategories( $userId, $categories );
+	}
+
+}
```

Comment:
This class can be removed. It does not add any value.

Having a UseCase also suggests that all app logic is in there. So this class currently suggests there is no logic such as permission checks, which is wrong, since you have a permission check in the SpecialPage.